From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 28 Mar 2016 14:31:07 +0200
Subject: [PATCH] {BUG in 4.6-rc1} scsi_dh_alua: fix uninitialized variable

drivers/scsi/device_handler/scsi_dh_alua.c:569:12: error: variable 'err' is used uninitialized whenever 'if'
      condition is false [-Werror,-Wsometimes-uninitialized]
                else if (sense_hdr.sense_key == UNIT_ATTENTION)
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/scsi/device_handler/scsi_dh_alua.c:571:7: note: uninitialized use occurs here
                if (err == SCSI_DH_RETRY &&
                    ^~~
drivers/scsi/device_handler/scsi_dh_alua.c:569:8: note: remove the 'if' if its condition is always true
                else if (sense_hdr.sense_key == UNIT_ATTENTION)
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/scsi/device_handler/scsi_dh_alua.c:516:14: note: initialize the variable 'err' to silence this warning
        unsigned err, retval;
                    ^
                     = 0
---
 drivers/scsi/device_handler/scsi_dh_alua.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/device_handler/scsi_dh_alua.c b/drivers/scsi/device_handler/scsi_dh_alua.c
index a655cf29c16f..debf3ce89769 100644
--- a/drivers/scsi/device_handler/scsi_dh_alua.c
+++ b/drivers/scsi/device_handler/scsi_dh_alua.c
@@ -513,7 +513,7 @@ static int alua_rtpg(struct scsi_device *sdev, struct alua_port_group *pg)
 	struct alua_port_group *tmp_pg;
 	int len, k, off, valid_states = 0, bufflen = ALUA_RTPG_SIZE;
 	unsigned char *desc, *buff;
-	unsigned err, retval;
+	unsigned err = 0 /* BUG */, retval;
 	unsigned int tpg_desc_tbl_off;
 	unsigned char orig_transition_tmo;
 	unsigned long flags;
-- 
